internal/build/buildconfig/maker.go

```diff
@@ -0,0 +1,124 @@
+package buildconfig
+
+import (
+	"fmt"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	kmmv1beta1 "github.com/rh-ecosystem-edge/kernel-module-management/api/v1beta1"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/build"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+//go:generate mockgen -source=maker.go -package=buildconfig -destination=mock_maker.go
+
+type Maker interface {
+	MakeBuildConfig(mod kmmv1beta1.Module, buildConfig *kmmv1beta1.Build, targetKernel, containerImage string) (*buildv1.BuildConfig, error)
+}
+
+type maker struct {
+	helper build.Helper
+	scheme *runtime.Scheme
+}
+
+func NewMaker(helper build.Helper, scheme *runtime.Scheme) Maker {
+	return &maker{
+		helper: helper,
+		scheme: scheme,
+	}
+}
+
+func (m *maker) MakeBuildConfig(mod kmmv1beta1.Module, buildConfig *kmmv1beta1.Build, targetKernel, containerImage string) (*buildv1.BuildConfig, error) {
```

Comment:
I think that `buildConfig` is a confusing name, at first sight, it make me think this is an OCP buildConfig object but indeed it is a KMM build object.

Reply:
Sure.

Reply:
Ended up using `kmmBuild` as `build` is a package name already.